### PR TITLE
WT-10550 Add an abort option to the verify utility

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

File: src/docs/command-line.dox [hunk 1]
```diff
@@ -472,15 +472,20 @@ Check the structural integrity of a table.
 The \c verify command verifies the specified table if a URI argument is provided,
 exiting success if the data source is correct, and failure if the data source is corrupted.
 If no URI argument is provided, it verifies each table in the database, reporting on any
-that are damaged. Currently, the verify operation is not supported with tiered tables
-and the history store file.
+that are damaged. If the abort on error option \c -a is specified, terminate the verification
+process if one of tables is damaged. Currently, the verify operation is not supported with
```
Comment (etienneptl) [inline src/docs/command-line.dox:476]: There might be a missing word:
one of *the* tables is damaged.

File: test/suite/test_verify.py [hunk 2]
```diff
@@ -302,5 +302,18 @@ def test_verify_all(self):
 
         self.runWt(["verify"])
 
+        # Purposely corrupt the last two tables. Test that verifying the database
+        # with the abort option stops after seeing the first corrupt table.
```
Comment (etienneptl) [inline test/suite/test_verify.py:306]: Typo: corrupted*

**Review by etienneptl — APPROVED** [review]
LGTM!

File: src/utilities/util_verify.c [hunk 3]
```diff
@@ -61,14 +61,16 @@ util_verify(WT_SESSION *session, int argc, char *argv[])
...
-    do_not_clear_txn_id = dump_address = dump_app_data = dump_blocks = dump_layout = dump_pages =
-      read_corrupt = stable_timestamp = false;
+    abort_on_error = do_not_clear_txn_id = dump_address = dump_app_data = dump_blocks =
+      dump_layout = dump_pages = read_corrupt = stable_timestamp = false;
     config = dump_offsets = uri = NULL;
-    while ((ch = __wt_getopt(progname, argc, argv, "cd:stu?")) != EOF)
+    while ((ch = __wt_getopt(progname, argc, argv, "acd:stu?")) != EOF)
         switch (ch) {
+        case 'a':
+            abort_on_error = true;
```
Comment (keitharnoldsmith) [inline src/utilities/util_verify.c:73]: Missing `break;`

**Review by keitharnoldsmith — APPROVED** [review]
One small bug, but otherwise LGTM!

File: src/utilities/util_verify.c (hunk 3 above)
Comment (mm-ng) [reply src/utilities/util_verify.c:73]: Added a break statement in the second commit.

Comment (keitharnoldsmith) [reply src/utilities/util_verify.c:73]: Ugh! I could have sworn I was looking at the "all commits" view. But apparently not...

**Pull request merged**